qiskit/primitives/base_estimator.py

```diff
@@ -0,0 +1,199 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+r"""
+=========
+Estimator
+=========
+
+Estimator class estimates expectation values of quantum circuits and observables.
+
+An estimator object is initialized with multiple quantum circuits and observables
+and users can specify pairs of quantum circuits and observables
+to estimate the expectation values.
+
+The input consists of following elements.
+
+* quantum circuits (:math:`\psi_i(\theta)`): list of (parametrized) quantum circuits
+  or state vectors to be converted into quantum circuits.
+  (a list of :class:`~qiskit.circuit.QuantumCircuit`))
```

Comment:
The type hint says only QuantumCircuit, not statevectors.
Also if we allow statevectors then we are requiring a synthesis of it to circuits, which might be hard especially for a parameterized statevector. 

Maybe the Aer & quantum_info implementation can allow it, but not the base abstract interface.

Reply:
Let's keep as circuit only for now